visualizer/src/service/tools/segment/floodMachine.js

```diff
@@ -1,40 +1,39 @@
 import { Machine, sendParent } from 'xstate';
 import { toolActions, toolGuards } from './toolUtils';
 
-const createFloodMachine = ({ x, y, label, foreground, background }) =>
-  Machine(
-    {
-      context: {
-        x,
-        y,
-        label,
-        foreground,
-        background,
-      },
-      on: {
-        COORDINATES: { actions: 'setCoordinates' },
-        LABEL: { actions: 'setLabel' },
-        FOREGROUND: { actions: 'setForeground' },
-        BACKGROUND: { actions: 'setBackground' },
-        mouseup: [{ cond: 'onBackground', actions: 'flood' }, { actions: 'selectBackground' }],
-      },
+const floodMachine = Machine(
+  {
+    context: {
+      x: null,
+      y: null,
+      label: null,
+      foreground: null,
+      background: null,
     },
-    {
-      guards: toolGuards,
-      actions: {
-        ...toolActions,
-        selectBackground: sendParent('SELECT_BACKGROUND'),
-        flood: sendParent(({ foreground, x, y }, event) => ({
-          type: 'EDIT',
-          action: 'flood',
-          args: {
-            label: foreground,
-            x_location: x,
-            y_location: y,
-          },
-        })),
-      },
-    }
-  );
+    on: {
+      COORDINATES: { actions: 'setCoordinates' },
+      HOVERING: { actions: 'setHovering' },
+      FOREGROUND: { actions: 'setForeground' },
+      BACKGROUND: { actions: 'setBackground' },
+      mouseup: { actions: ['flood', 'selectBackground'] },
```

Comment:
yes this was one of the changes I noted in the top level PR description.

> fixing the flood tool to flood immediately instead of first selecting the background and then flooding